index.js

```diff
@@ -83,15 +88,14 @@ function connect (appKey, accessToken, _opts) {
       return
     }
 
-    response
-      .pipe(zlib.createGunzip())
-        .on('error', emitError)
-      .pipe(split(parseJSON))
-        .on('data', emitData)
-        .on('end', checkReconnect)
+    responseStream = response
+    pump(response, zlib.createGunzip(), split(parseJSON))
```

Comment:
switched to using `pump` here so that ending any stream in the chain will unpipe everything properly.